src/vsr/client.zig

```diff
@@ -433,6 +446,32 @@ pub fn ClientType(comptime StateMachine_: type, comptime MessageBus: type) type
                 });
                 self.view = pong.header.view;
             }
+
+            const ping_timestamp_monotonic = pong.header.ping_timestamp_monotonic;
+            const pong_timestamp_monotonic = self.time.monotonic();
+            if (ping_timestamp_monotonic <= pong_timestamp_monotonic) {
+                self.replica_round_trip_times_ns[pong.header.replica] =
+                    pong_timestamp_monotonic - ping_timestamp_monotonic;
+
+                var round_trip_times_ns = stdx.BoundedArrayType(u64, constants.replicas_max){};
+                for (self.replica_round_trip_times_ns) |round_trip_time_ns| {
+                    if (round_trip_time_ns) |rtt_ns| {
+                        round_trip_times_ns.append_assume_capacity(rtt_ns);
+                    }
+                }
+                std.mem.sort(u64, round_trip_times_ns.slice(), {}, std.sort.asc(u64));
+
+                const rtt_median_ns =
+                    round_trip_times_ns.get(@divFloor(round_trip_times_ns.count(), 2));
+                self.request_timeout.set_rtt_ns(rtt_median_ns);
+            } else {
+                log.debug("{}: on_pong: monotonic timestamp regressed {}..{} replica={}", .{
+                    self.id,
+                    ping_timestamp_monotonic,
+                    pong_timestamp_monotonic,
+                    pong.header.replica,
+                });
+            }
         }
```

Comment:
We repeat this "get median of things" code two times, and it would be great to have `stdx.median` API here, but I don't see how that could work, sadly... The problem is that we need to filter-out null elements from the input slice first, and that's tricky to express without iterators. 

So perhaps just repeating it twice is best!

Reply:
Yeah, also I think this is temporary -- as we add more "topology-awareness" I think we will come up with a better approach anyway.